applications/luci-app-simple-adblock/htdocs/luci-static/resources/simple-adblock/status.js

```diff
@@ -0,0 +1,306 @@
+					]);
+					return RPC.setInitAction(pkg.Name, 'start');
+				}
+			}, _('Start'));
+
+			var btn_action = E('button', {
+				'class': 'btn cbi-button cbi-button-apply',
+				disabled: true,
+				click: function (ev) {
+					ui.showModal(null, [
+						E('p', { 'class': 'spinning' }, _('Force re-downloading %s block lists').format(pkg.Name))
+					]);
+					return RPC.setInitAction(pkg.Name, 'dl');
+				}
+			}, _('Force Re-Download'));
+
+			var btn_stop = E('button', {
+				'class': 'btn cbi-button cbi-button-reset',
+				disabled: true,
+				click: function (ev) {
+					ui.showModal(null, [
+						E('p', { 'class': 'spinning' }, _('Stopping %s service').format(pkg.Name))
+					]);
+					return RPC.setInitAction(pkg.Name, 'stop');
+				}
+			}, _('Stop'));
+
+			var btn_enable = E('button', {
+				'class': 'btn cbi-button cbi-button-apply',
+				disabled: true,
+				click: function (ev) {
+					ui.showModal(null, [
+						E('p', { 'class': 'spinning' }, _('Enabling %s service').format(pkg.Name))
+					]);
+					return RPC.setInitAction(pkg.Name, 'enable');
+				}
+			}, _('Enable'));
+
+			var btn_disable = E('button', {
+				'class': 'btn cbi-button cbi-button-reset',
+				disabled: true,
+				click: function (ev) {
+					ui.showModal(null, [
+						E('p', { 'class': 'spinning' }, _('Disabling %s service').format(pkg.Name))
+					]);
+					return RPC.setInitAction(pkg.Name, 'disable');
+				}
+			}, _('Disable'));
+
+			if (replyStatus[pkg.Name].enabled) {
+				btn_enable.disabled = true;
+				btn_disable.disabled = false;
+				switch (replyStatus[pkg.Name].status) {
+					case 'statusSuccess':
+						btn_start.disabled = true;
+						btn_action.disabled = false;
+						btn_stop.disabled = false;
+						break;
+					case 'statusStopped':
+						btn_start.disabled = false;
+						btn_action.disabled = true;
+						btn_stop.disabled = true;
+						break;
+					default:
+						btn_start.disabled = true;
+						btn_action.disabled = true;
+						btn_stop.disabled = true;
+						btn_enable.disabled = true;
+						btn_disable.disabled = true;
+					break;
+				}
+			}
+			else {
+				btn_start.disabled = true;
+				btn_action.disabled = true;
+				btn_stop.disabled = true;
+				btn_enable.disabled = false;
+				btn_disable.disabled = true;
+			}
+
+			var buttonsDiv = [];
+			var buttonsTitle = E('label', { class: 'cbi-value-title' }, _("Service Control"))
+			var buttonsText = E('div', {}, [btn_start, btn_gap, btn_action, btn_gap, btn_stop, btn_gap_long, btn_enable, btn_gap, btn_disable]);
+			var buttonsField = E('div', { class: 'cbi-value-field' }, buttonsText);
+			if (replyStatus[pkg.Name].version) {
+				buttonsDiv = E('div', { class: 'cbi-value' }, [buttonsTitle, buttonsField]);
+			}
+
+			return E('div', {}, [header, statusDiv, warningsDiv, errorsDiv, buttonsDiv]);
+		});
+	},
+});
+
+RPC.on('setInitAction', function (reply) {
+	ui.hideModal();
+	location.reload();
```

Comment:
If there are any other async operations they might not complete in time before the page reloads, but I guess that doesn't matter as much...